src/PrestaShopBundle/Bridge/AdminController/ControllerConfigurationFactory.php

```diff
@@ -51,28 +51,57 @@ public function __construct(
     }
 
     /**
-     * @param int $id
-     * @param string $controllerName
-     * @param string $controllerNameLegacy
-     * @param string $table
+     * @param int $tabId
+     * @param string $objectModelClassName
+     * @param string $legacyControllerName
+     * @param string $tableName
      *
      * @return ControllerConfiguration
      */
     public function create(
-        int $id,
-        string $controllerName,
-        string $controllerNameLegacy,
-        string $table
+        int $tabId,
+        string $objectModelClassName,
+        string $legacyControllerName,
+        string $tableName
     ): ControllerConfiguration {
-        $configuratorController = new ControllerConfiguration();
+        $controllerConfiguration = new ControllerConfiguration();
+        $controllerConfiguration->tabId = $tabId;
+        $controllerConfiguration->objectModelClassName = $objectModelClassName;
+        $controllerConfiguration->legacyControllerName = $legacyControllerName;
+        $controllerConfiguration->tableName = $tableName;
+        $controllerConfiguration->user = $this->userProvider->getUser();
+        $controllerConfiguration->templateFolder = Tools::toUnderscoreCase(substr($controllerConfiguration->legacyControllerName, 5)) . '/';
```

Comment:
Same for the currentIndex and the token, instead of having some setters method you can turn those into getters methods that will be used for the constructor's parameters